sections/Deals/CreateDeal.tsx

```diff
@@ -264,6 +264,14 @@ const CreateDeal: FC<CreateDealProps> = ({ poolAddress, purchaseToken }) => {
 		};
 	}, [formik?.values, poolAddress, provider, signer, walletAddress]);
 
+	const underlyingDealTokenTotal = useMemo(() => {
+		if (!formik.values?.purchaseTokenTotal || formik.values.dealExchangeRate <= 0) {
+			return 0;
+		}
+
+		return Number(wei(formik.values.dealExchangeRate).mul(wei(formik.values.purchaseTokenTotal)));
```

Comment:
I am not sure but I think .toNumber() is safer with decimal precision than using the Number() method

Reply:
Done